pgdog/src/frontend/router/parser/query.rs

```diff
@@ -226,6 +244,13 @@ impl QueryParser {
         let mut command = match root.node {
             // SELECT statements.
             Some(NodeEnum::SelectStmt(ref stmt)) => {
+                // Check for side-effect functions in the target list
+                if Self::has_side_effect_function(stmt) {
```

Comment:
This is called a bit too early. Right below is where we check the comment for a manually selected shard. You want to support use cases like:

```sql
/* pgdog_shard: 0 */ SELECT nextval('sequence_name');
```